pkg/morph/client/notifications.go

```diff
@@ -235,11 +235,26 @@ routeloop:
 		select {
 		case <-c.closeChan:
 			break routeloop
-		case ev, ok := <-conn.notifyChan:
+		case ev, ok := <-func() <-chan *state.ContainedNotificationEvent {
```

Comment:
Obfuscated Go contest?

This code was specifically written with `conn == nil` expected to be handled above. `c.switchRPC()` can not return `nil` `conn`. Then `conn` may be outdated by the time we get here, but it's not `nil`.

Reply:
OK, I see now it can. This happens in exactly one case and it can be handled right in `switchRPC()` result handling. Just break out of the `routeloop` immediately.

Reply:
In fact, it's all 5ec3d5a1762e3dabc287fdd0d535306c9b38d278 error. The original intention was to ensure `conn != nil`, but it failed for `closeChan` case really.

Reply:
> it can be handled right in `switchRPC()` result handling. Just break out of the `routeloop` immediately.

Yes, it seems to be working. There is no more panic.